### ISSUE TITLE
 TypeError: Cannot read property 'invalidateSize' of null

### ISSUE THREAD

**Issue by skyqrose** [krvital/react-leaflet-fullscreen#9]
When clicking the control to go into or out of fullscreen, the error fires 3 or 4 times. `map` is `null`.

Unlike #6, this does occur in production, though it's not causing any user-facing problems beyond filling up the console with error messages.

Earlier today, it was happening every time I clicked the fullscreen control. Now it's not happening at all. I'm not sure what causes it to happen or not.

react-leaflet-fullscreen version 1.0.1
leafelt.fullscreen version 1.4.5

```ts
// node_modules/leaflet.fullscreen/Control.FullScreen.js line 89
_handleFullscreenChange: function () {
	var map = this._map;
	map.invalidateSize();
	if (!fullScreenApi.isFullScreen() && !map._exitFired) {
		map.fire('exitFullscreen');
		map._exitFired = true;
		map._isFullscreen = false;
	}
```

**Comment by dansoliveira** [comment]
Hi!

I'm having the same problem.

Basically, every time the component is rendered, function `onAdd` is called, consequently the `_createButton`, responsibly to add listener to `_handleFullscreenChange` on `document`.
But the listener isn't removed. So, every we click on fullscreen button, this function is called to on invalid context that map doesn't exists.

This problem occurs on [leaflet.fullscreen](https://github.com/brunob/leaflet.fullscreen) dependency, and was fixed on v1.6.0 through [this commit](https://github.com/brunob/leaflet.fullscreen/commit/4da2a5a5a1bc94d81a672e140f6acb209dc071b0).

I'll test with this update and create an PR to fix this.

**Comment by krvital** [comment]
Hey guys! Check the new 1.x version

**Comment by skyqrose** [comment]
Do you have a changelog? What's different between 1.0.2 and 2.0.2?

**Comment by krvital** [comment]
You are right I should have added a changelog, thanks for the advice. I'll do it later.

I kept 1.x version for compatibility with react-leaflet 2.x. The 2.x version of react-leaflet-fullscreen should be used with react-leaflet 3.x
